master/internal/configpolicy/utils.go

```diff
@@ -81,11 +81,36 @@ func ValidateExperimentConfig(
 		return err
 	}
 
+	// Warn the user when fields specified in workspace config policies overlap with global config
+	// policies (since these fields will be overridden by the respective fields in the global
+	// policies).
+	var globalConstraints *model.Constraints
+	var globalConfig *expconf.ExperimentConfig
 	if globalConfigPolicies != nil {
-		checkAgainstGlobalConfig[model.Constraints](globalConfigPolicies.Constraints, cp.Constraints, "invalid constraints")
-		checkAgainstGlobalConfig[expconf.ExperimentConfig](
-			globalConfigPolicies.InvariantConfig, cp.InvariantConfig, InvalidExperimentConfigPolicyErr,
-		)
+		if globalConfigPolicies.Constraints != nil {
+			globalConstraints, err = UnmarshalConfigPolicy[model.Constraints](
+				*globalConfigPolicies.Constraints,
+				InvalidExperimentConfigPolicyErr,
+			)
+			if err != nil {
+				ConfigPolicyWarning(err.Error())
+				return err
+			}
+		}
+
+		if globalConfigPolicies.InvariantConfig != nil {
+			globalConfig, err = UnmarshalConfigPolicy[expconf.ExperimentConfig](
+				*globalConfigPolicies.InvariantConfig,
+				InvalidExperimentConfigPolicyErr,
+			)
+			if err != nil {
+				ConfigPolicyWarning(err.Error())
+				return err
+			}
+		}
```

Comment:
This unmarshal logic is not repeated, right? Let's abstract it into a function.

Reply:
Just making sure, you want me to create an additional `UnmarshalExperimentConfigPolicy` and `UnmarshalNTSCConfigPolicy` like the ones we had [before](https://github.com/determined-ai/determined/pull/10028/files#diff-23e49cf55e746838ac168bf9c749b1f2185e72e40114b5dde5a85d77318c8b9dL35-L72), except they take in optional `*string` rather than `string`?

Reply:
This was a great idea, code looks much cleaner!